src/main/compile/middle/Backifier.scala

```diff
@@ -0,0 +1,113 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.compile
+package middle
+
+import org.nlogo.core.{Instantiator, BreedIdentifierHandler, Program}
+import org.nlogo.{ api, core, nvm, prim },
+
+  nvm.Procedure.ProceduresMap
+
+// This is seriously gross and horrible. - ST 4/11/14
+
+class Backifier(
+  program: Program,
+  extensionManager: api.ExtensionManager,
```

Comment:
By the time we get here we're assuming that each errormessage has been paired with a carefully, so any that haven't are errors in the AST. The stage that does the pairing checks to make sure that all are matched, so we shouldn't get here with any having a let of None. I can add a case with an error message to make that explicit, if you would like.